SDWebImage/SDImageCache.h

```diff
@@ -54,12 +54,32 @@ typedef void(^SDWebImageCompletionWithPossibleErrorBlock)(NSError * _Nullable er
 #pragma mark - Properties
 
 /**
- *  Cache Config object - storing all kind of settings
+ *  Cache Config object - storing all kind of settings.
+ *  The property is copy so change of currrent config will not accidentally affect other cache's config.
  */
-@property (nonatomic, nonnull, readonly) SDImageCacheConfig *config;
+@property (nonatomic, copy, nonnull, readonly) SDImageCacheConfig *config;
```

Comment:
Actually I think it's a issue by design. The configuration for class which contains a shared instance should always use copy. Because when you change the config on shared instance, this may affect other instance who use the same config object.

Reply:
@skyline75489 Because as I point it out. We now have a default configuration in static. This means each `SDImageCache` instance will call `copy` to get a copy and store in their own. Or if you modify the default configuration, all the `SDImageCache` instance's config will be affect, this is a wrong behavoir.

Another way, it's to change the default value for `SDImageCacheConfig`. Each time you call `[[SDImageCacheConfig alloc] init]`, the config contains all the default value. However, this is hard to use because if I want to specify the `sharedImageCache` to use the custom `class<SDDiskCache>`. I have no proper timing to do this. Right ? After you call `[SDImageCache sharedCache]`, you already use the default class intead of my own class.